x-pack/plugin/rank-rrf/src/main/java/org/elasticsearch/xpack/rank/rrf/RRFRankBuilder.java

```diff
@@ -55,6 +60,13 @@ public class RRFRankBuilder extends RankBuilder {
     }
 
     public static RRFRankBuilder fromXContent(XContentParser parser) throws IOException {
+        if (parser.getRestApiVersion().matches(RestApiVersion.onOrAfter(RestApiVersion.V_8))) {
```

Comment:
So, this was never available in any previous rest API versions. I don't think this is required. 

Also, We should just deprecate the constructor. I think for parsing, the deprecation belongs in the SearchSourceBuilder as that is the only place that actually uses this right?